pay-api/src/pay_api/services/payment_account.py

```diff
@@ -12,155 +12,278 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Service to manage Payment Account model related operations."""
+from __future__ import annotations
 
-from typing import Any, Dict
+from typing import Any, Dict, Union
 
 from flask import current_app
 
-from pay_api.exceptions import BusinessException
 from pay_api.models import PaymentAccount as PaymentAccountModel
 from pay_api.models import StatementSettings as StatementSettingsModel
-from pay_api.models.bcol_payment_account import BcolPaymentAccount
-from pay_api.models.credit_payment_account import CreditPaymentAccount
-from pay_api.models.internal_payment_account import InternalPaymentAccount
-from pay_api.utils.enums import PaymentSystem, StatementFrequency
-from pay_api.utils.errors import Error
+from pay_api.models import CfsAccount as CfsAccountModel
+from pay_api.utils.enums import StatementFrequency, AccountType, PaymentMethod
 from pay_api.utils.util import get_str_by_path
-from pay_api.utils.user_context import user_context, UserContext
 
 
 class PaymentAccount():  # pylint: disable=too-many-instance-attributes
     """Service to manage Payment Account model related operations."""
 
     def __init__(self):
         """Initialize service."""
-        self._id: int = None
-        self._corp_number: str = None
-        self._corp_type_code: str = None
-        self._payment_system_code: str = None
-        self._paybc_account: str = None
-        self._paybc_party: str = None
-        self._paybc_site: str = None
-        self._bcol_user_id: str = None
-        self._bcol_account_id: str = None
-        self._account_id: str = None
-
-    def populate(self, value):
-        """Populate."""
-        if value:
-            self._id: int = value.id
-            if isinstance(value, BcolPaymentAccount):
-                self._payment_system_code = PaymentSystem.BCOL.value
-                self._bcol_account_id = value.bcol_account_id
-                self._bcol_user_id = value.bcol_user_id
-                self._account_id = value.account_id
-            elif isinstance(value, InternalPaymentAccount):
-                self._payment_system_code = PaymentSystem.INTERNAL.value
-                self._corp_number = value.corp_number
-                self._corp_type_code = value.corp_type_code
-                self._account_id = value.account_id
-            elif isinstance(value, CreditPaymentAccount):
-                self._payment_system_code = PaymentSystem.PAYBC.value
-                self._paybc_account = value.paybc_account
-                self._paybc_party = value.paybc_party
-                self._paybc_site = value.paybc_site
-                self._corp_number = value.corp_number
-                self._corp_type_code = value.corp_type_code
-                self._account_id = value.account_id
+        self.__dao = None
+        self._id: Union[None, int] = None
```

Comment:
I guess https://docs.python.org/3/library/typing.html#typing.Optional can be used here ; havent tried this ,but looks like optional for this purpose

Reply:
Went with Union, as this is in initialization of instance attributes. Ideally those values will get populated right after initialization. I believe Optional is more appropriate in definitions were the value can be None or a type. For initialization of instance attributes I am not sure if Optional is the correct one to use.